### DISCUSSION TITLE
How to allow my application to read machine serial number?

### DISCUSSION THREAD

**Discussion by calve** [netblue30/firejail#5164] [Q&A] [answered]
Hi

not sure if this is a bug or a feature but I cant get the right configuration to allow an application to read the hardware serial number and other informations living in `/sys/firmware/`

```
$ ls -alh /sys/firmware 
total 0
drwxr-xr-x  6 root root 0 25 mai   14:13 .
dr-xr-xr-x 13 root root 0 25 mai   14:13 ..
drwxr-xr-x  7 root root 0 25 mai   14:13 acpi
drwxr-xr-x  4 root root 0 25 mai   14:13 dmi
drwxr-xr-x  5 root root 0 25 mai   14:13 efi
drwxr-xr-x 16 root root 0 27 mai   15:33 memmap

$ sudo firejail --noprofile ls -alh /sys/firmware

Parent pid 105126, child pid 105127
The new log directory is /proc/105127/root/var/log
Child process initialized in 7.60 ms
total 0
dr--------  2 root root 40 27 mai   15:30 .
dr-xr-xr-x 13 root root  0 25 mai   14:13 ..

Parent is shutting down, bye...
```

what am I missing here ?

I understand `/sys` is mounted as `sysfs` on the host. 
The firejail-profiles manpage specifically mention that  `allow` work for all but `/proc` and `/sys` and does not give reasons. The firejail manpag let me interpret that I must start my kernel with apparmor ?

**Comment by rusty-snake** [comment, marked as answer]
> a bug or a feature

That's one of the biggest questions in software development. If in doubt it's always a feature :smile:.

>  I cant get the right configuration to allow an application to read the hardware serial number and other informations living in `/sys/firmware`

ATM this is impossible (without rebuilding firejail). The `blacklist` is unconditional done.

https://github.com/netblue30/firejail/blob/baeec77b760cb7789dba99f885d0d65a817a335b/src/firejail/fs.c#L741

Perhaps we should allow `noblacklist /sys/firmware` like we do for `/sys/fs` and `/sys/module`.

> what am I missing here ?

General: Always try with `--profile=noprofile` if `--noprofile` does not work. (here it will not help)

**Reply by calve** [reply]
Thanks this make sense. Indeed, if I recompile firejail after commenting the line you point, I can successfully read my hardware serial inside firejail.

> Perhaps we should allow noblacklist /sys/firmware like we do for /sys/fs and /sys/module.

I'd advocate yes but what would be the reason to not ?

**Reply by rusty-snake** [reply]
> I'd advocate yes but what would be the reason to not ?

Likely nothing (except for performance).
